api/base/serializers.py

```diff
@@ -1003,7 +1009,19 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class JSONAPISerializer(ser.Serializer):
+class PrefetchRelationshipsSerializer(ser.Serializer):
+    relationship_field_names = list()
+
+    def __init__(self, *args, **kwargs):
+        super(PrefetchRelationshipsSerializer, self).__init__(*args, **kwargs)
+        for field_name, field in self.fields.iteritems():
+            if field.source == '*':
+                self.relationship_field_names.append(field_name)
```

Comment:
> Also, PrefetchRelationshipsSerializer doesn't have any prefetching behavior, so we can probably get rid of it and just put this code on JSONAPISerializer.

Happy to rename it but there are a relatively large amount of serializers that don't inherit from the JSONAPISerializer and I didn't want to deal with the reasons/repercussions of changing that.